### PR TITLE
[DataGrid] Forward ref to root element in `GridEditInputCell`

### LINKED ISSUE #5457: [data grid] Warnings with the Validation Demo in the docs
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Implementing [this example](https://mui.com/x/react-data-grid/editing/#validation) produces a couple of React errors.

> Warning: Failed prop type: Invalid prop `children` supplied to `ForwardRef(Tooltip2)`. Expected an element that can hold a ref. Did you accidentally use a plain function component for an element instead? For more information see https://mui.com/r/caveat-with-refs-guide

> Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
Check the render method of `ForwardRef(Tooltip2)`.
    at GridEditInputCell 

> MUI: The `children` component of the Tooltip is not forwarding its props correctly.
Please make sure that props are spread on the same element that the ref is applied to. 
    at Tooltip2

Things I've tried:
- I initially got this error when replacing the `DataGridPro` components with the standard `DataGrid` component but later confirmed that I still received the error with `DataGridPro` as well
- Read through the `caveat-with-refs-guide` link provided in the error log but as far as I can tell, this should be valid? 
- My own implementation that simplifies the check a bit with local variables / doesn't use a promise
- Using a stock Tooltip component instead of a styled
- Wrapping the component in Fragments, this removed the error but resulted in the Tooltip not showing at all
- I used a stock MUI `Input` component instead of the `GridEditInputCell` component, no error was produced but not ideal

### Expected behavior 🤔

No React errors produced.

### Steps to reproduce 🕹

Steps:
1. Run the [example code](https://mui.com/x/react-data-grid/editing/#validation) from the docs with non-production React
2. Double click a cell to edit it
3. See errors in browser console

### Context 🔦

I'm trying to setup a validation system, if there's a better way to do this than the examples in the documentation then I'm open to alternatives!

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
Using Chrome for this specific example
  System:
    OS: macOS 12.2.1
  Binaries:
    Node: 16.14.2 - /usr/local/bin/node
    Yarn: 1.23.0-20220130.1630 - /opt/homebrew/bin/yarn
    npm: 8.5.5 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 103.0.5060.114
    Edge: Not Found
    Firefox: 91.7.1
    Safari: 15.3
  npmPackages:
    @emotion/react:  11.9.0 
    @emotion/styled:  11.8.1 
    @mui/base:  5.0.0-alpha.82 
    @mui/icons-material:  5.8.4 
    @mui/material:  5.8.1 
    @mui/private-theming:  5.8.0 
    @mui/styled-engine:  5.8.0 
    @mui/system:  5.8.1 
    @mui/types:  7.1.3 
    @mui/utils:  5.8.0 
    @mui/x-data-grid:  5.13.0 
    @mui/x-data-grid-pro:  5.13.0 
    @mui/x-license-pro:  5.12.3 
    @types/react:  17.0.45 
    react:  17.0.2 
    react-dom:  17.0.2 
    typescript: ^4.4.3 => 4.6.4 ```
</details>


### Order ID 💳 (optional)

_No response_

### LINKED ISSUE #5630: [data grid] GridEditInputCell should forward ref
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

When the validation method returns an error, those errors show up in the console:

### Expected behavior 🤔

There shouldn't be any errors.

### Steps to reproduce 🕹

Link to live example: https://codesandbox.io/s/jv30c0?file=/demo.tsx

Steps:
1. Input a name that already exists
2. Look at the console

### Context 🔦

I'm implementing validators in my app.

I fixed this issue by adding a `<Box>` that holds the edit component ref between the `Tooltip` and the `GridEditInputCell`

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  System:
    OS: Windows 10 10.0.19044
  Binaries:
    Node: 16.13.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.11 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.1.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: Version 103.0.5060.134 (Official Build) (64-bit)
    Edge: Spartan (44.19041.1266.0), Chromium (103.0.1264.71)
```
</details>

### Order ID 💳 (optional)

47285

### PR DESCRIPTION
**Pull request by Zenoo**
This should fix #5630

Closes #5457

### CODE REVIEW

File: packages/grid/x-data-grid/src/components/cell/GridEditInputCell.tsx [hunk 1]
```diff
@@ -197,6 +200,6 @@ GridEditInputCell.propTypes = {
 
 export { GridEditInputCell };
 
-export const renderEditInputCell = (params: GridEditInputCellProps) => (
-  <GridEditInputCell {...params} />
+export const renderEditInputCell = React.forwardRef<HTMLInputElement, GridEditInputCellProps>(
+  (params, ref) => <GridEditInputCell {...params} ref={ref} />,
```
Comment (m4theushw) [inline packages/grid/x-data-grid/src/components/cell/GridEditInputCell.tsx:204]: Could you revert this part? It's not needed. We never call `renderEditCell` passing a ref. To stop TypeScript from complaining that `ref` is not passed you also need to include `ref` to the list of omitted props that make `GridEditInputCellProps`. See https://github.com/mui/mui-x/issues/5457#issuecomment-1181037325

File: packages/grid/x-data-grid/src/components/cell/GridEditInputCell.tsx [hunk 2]
```diff
@@ -53,73 +53,76 @@ export interface GridEditInputCellProps
...
-  const classes = useUtilityClasses(ownerState);
-
-  const handleChange = React.useCallback(
-    async (event) => {
-      const newValue = event.target.value;
-
-      if (onValueChange) {
-        await onValueChange(event, newValue);
+const GridEditInputCell = React.forwardRef<HTMLInputElement, GridEditInputCellProps>(
+  (props: GridEditInputCellProps, ref) => {
```
Comment (m4theushw) [inline packages/grid/x-data-grid/src/components/cell/GridEditInputCell.tsx:57]: `props` is already typed by `React.forwardRef`.

```suggestion
  (props, ref) => {
```

**Review by m4theushw — COMMENTED** [review]
Thanks for the contribution.

We already received a report that the validation demo with the tooltip was emitting errors because the ref is not forwarded. I updated the description to close #5630 instead. Unfortunately, we won't be able to get rid of all errors due to a problem in the Tooltip (see https://github.com/mui/mui-x/issues/5457#issuecomment-1181037325) but it's already an improvement.

File: packages/grid/x-data-grid/src/components/cell/GridEditInputCell.tsx (hunk 1 above)
Comment (Zenoo) [reply packages/grid/x-data-grid/src/components/cell/GridEditInputCell.tsx:204]: Oh, I see, thanks !

**Comment by Zenoo** [comment]
Hum, running `yarn proptypes` is telling me to remove `isRequired` from every proptype of `GridEditInputCell`.  
That doesn't feel right though

**Comment by m4theushw** [comment]
> Hum, running yarn proptypes is telling me to remove isRequired from every proptype of GridEditInputCell.

You can go forward and commit the changes from `yarn proptypes`. This change is happening because we have an index signature in

https://github.com/mui/mui-x/blob/ed5d5632a297cb15f776ce098f27b5e1f78df86b/packages/grid/x-data-grid/src/models/gridEditRowModel.ts#L1-L6

and it doesn't play correctly with `React.forwardRef`. But fortunately the props inside `GridEditInputCell` are not treated as optional and this is what matters. In v6 we want to remove this index signature.

**Comment by Zenoo** [comment]
Thanks for the explanation, everything should be good now.

**Review by m4theushw — APPROVED** [review]

**Comment by cherniavskii** [comment]
I'm not sure the issue was fixed though: https://codesandbox.io/s/validateservernamegrid-demo-mui-x-gm3x8b?file=/package.json

**Comment by m4theushw** [comment]
@cherniavskii See https://github.com/mui/material-ui/issues/33476 What we could do here, forward the ref, we did.

**Pull request merged**